docs/pages/machine-id/introduction.mdx

```diff
@@ -0,0 +1,55 @@
+---
+title: Machine ID
+description: Teleport Machine ID introduction, demo and resources.
+---
+
+# Machine ID
+
+Machine ID is a fully automated Certificate Authority designed to
```

Comment:
I think we should elaborate either here or in the Getting Started page how Machine ID interacts with the Teleport Auth Service and Proxy Service. 

If a reader understands the Auth Service to be the cluster's certificate authority, for example, the statement that Machine ID is also a certificate authority could lead to some potential confusion.

Also, Machine ID introduces a fourth kind of Teleport service (in addition to the Auth, Proxy, and Nodes), so I think this should be clearer in terms of architecture about how that service runs (e.g., ports and network protocols).

Maybe we could even include an architectural diagram in this page that shows the Auth, Proxy, Nodes, and Machine ID? If that's out of scope for now, we should at least create an issue to track this for later.

Reply:
Hrm, you're right that Machine ID is not a CA itself, but it does get a bit technical to explain what it is. "Machine ID is a client that exposes Teleports internal CA".

I am planning on writing a architecture section after I write a few guides that might help explain how Machine ID sits within a cluster?

Reply:
Also note that we already have more than 3 types of services. In addition to auth, proxy, and node, we also have windows desktop service, app service, DB service, and Kubernetes service.

Reply:
@russjones That wording sounds good to me, as does eventually editing our architecture guides to talk about Machine ID. 

I also think it would be good to include one or two sentences here that mention architecture at a high level. Maybe the first paragraph could be something like,

```markdown
Machine ID is a daemon that programmatically issues and 
renews short-lived certificates to any service account 
(e.g., a CI/CD server) by retrieving credentials from the 
Teleport Auth Service. This enables fine-grained role-based
access controls and audit.
```